scheduler/setup_funcs.go

```diff
@@ -40,67 +37,6 @@ func PopulateCaches(id string, distroID string, tasks []task.Task) ([]task.Task,
 	return cmp.tasks, nil
 }
 
-// cacheTaskGroups caches task groups by version. It uses yaml.Unmarshal instead
-// of model.LoadProjectInto and only unmarshals task groups for efficiency.
-func cacheTaskGroups(comparator *CmpBasedTaskComparator) error {
-	comparator.projects = make(map[string]project)
-	for _, v := range comparator.versions {
-		fullP := &model.Project{}
-		if err := model.LoadProjectInto([]byte(v.Config), "", fullP); err != nil {
-			return errors.Wrapf(err, "error getting task groups from version %s", v.Id)
-		}
-		comparator.projects[v.Id] = project{TaskGroups: fullP.TaskGroups}
-	}
-	return nil
-}
-
-func backfillTaskGroups(comparator *CmpBasedTaskComparator) error {
```

Comment:
Should we keep this for versions that still don't have it set, but have it noop?